Description: Fix heap overread in xsltFormatNumberConversion
 An empty decimal-separator could cause a heap overread. This can be
 exploited to leak a couple of bytes after the buffer that holds the
 pattern string.
 .
 Found with afl-fuzz and ASan.
 .
 This is CVE-2016-4738
Author: Nick Wellnhofer <wellnhofer@aevum.de>
Bug-Debian: https://bugs.debian.org/842570
Bug-Debian: https://security-tracker.debian.org/tracker/CVE-2016-4738
Last-Update: 2016-10-30

--- a/libxslt/numbers.c
+++ b/libxslt/numbers.c
@@ -1099,7 +1099,8 @@
     }
 
     /* We have finished the integer part, now work on fraction */
-    if (xsltUTF8Charcmp(the_format, self->decimalPoint) == 0) {
+    if ( (*the_format != 0) &&
+         (xsltUTF8Charcmp(the_format, self->decimalPoint) == 0) ) {
         format_info.add_decimal = TRUE;
 	the_format += xsltUTF8Size(the_format);	/* Skip over the decimal */
     }
